spec/requests/campaigns_request_spec.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Campaigns API', type: :request do
+  before do
+    @seller = create :seller
+    @location = create(:location, seller_id: @seller.id)
+    @campaign = create(:campaign, seller_id: @seller.id, location_id: @location.id)
+  end
+
+  context 'GET /campaigns' do
+    before { get '/campaigns' }
+
+    it 'Returns campaigns' do
+      expect(json).not_to be_empty
+      expect(json.size).to eq(1)
+    end
+
+    it 'Returns 200' do
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  context 'GET /campaigns/:id' do
+    before { get "/campaigns/#{campaign_id}" }
+
+    context 'With a missing id' do
+      let(:campaign_id) { 'missing_id' }
+
+      it 'Returns 404' do
+        expect(response).to have_http_status(404)
+      end
+    end
+
+    context 'With valid id' do
+      let(:campaign_id) { @campaign.id }
+
+      it 'Returns the campaign' do
+        expect(json).not_to be_empty
+        expect(json['id']).to eq(@campaign.id)
+      end
+
+      it 'Returns 200' do
+        expect(response).to have_http_status(200)
+      end
+    end
+  end
+
+  context 'POST /campaigns' do
+    context 'With invalid parameters' do
```

Comment:
Should we test invalid seller_id and location_id?

Reply:
Good call, added